Marlin/src/gcode/parser.h

```diff
@@ -309,13 +309,59 @@ class GCodeParser {
     }
 
     static inline float axis_unit_factor(const AxisEnum axis) {
-      return (
-        #if HAS_EXTRUDERS
-          axis >= E_AXIS && volumetric_enabled ? volumetric_unit_factor : linear_unit_factor
-        #else
-          linear_unit_factor
-        #endif
-      );
+      float unit_factor;
+      #if HAS_EXTRUDERS
+        if (axis >= E_AXIS && volumetric_enabled
+          #if HAS_ROTATIONAL_AXES
+            && axis != I_AXIS
+            #if HAS_ROTATIONAL_AXIS5
+              && axis != J_AXIS
+            #endif
+            #if HAS_ROTATIONAL_AXIS6
+              && axis != K_AXIS
+            #endif
+            #if HAS_ROTATIONAL_AXIS7
+              && axis != U_AXIS
+            #endif
+            #if HAS_ROTATIONAL_AXIS8
+              && axis != V_AXIS
+            #endif
+            #if HAS_ROTATIONAL_AXIS9
+              && axis != W_AXIS
+            #endif
+          #endif
+        ) {
+          unit_factor = volumetric_unit_factor;
+        }
+      #endif
+      #if HAS_ROTATIONAL_AXES
+        if (axis == I_AXIS
+          #if HAS_ROTATIONAL_AXIS5
```

Comment:
Is there a rule that if AXIS5 is rotational AXIS4 also has to be rotational?

From your config changes I thought they could all be configured independently, making the assumption that I_AXIS is rotational invalid.

Reply:
The code indeed expects that rotational axes come before secondary linear axes because it simplified code.  Until recently, axes were automatically set to rotational for AXIS*_NAME A/B/C. Now that we allow setups with more than 3 rotational axes, I will have to add sanity checks and documentation to enforce this. thx

Reply:
sanity checks and documentation are in place now